android/src/main/java/cl/json/social/MessengerShare.java

```diff
@@ -0,0 +1,45 @@
+package cl.json.social;
+
+import android.content.ActivityNotFoundException;
+import android.content.Intent;
+import java.io.File;
+import android.os.Environment;
+import android.net.Uri;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReadableMap;
+
+/**
+ * Created by Muhzi4u on 14-01-19.
+ */
+public class MessengerShare extends SingleShareIntent {
+
+    private static final String PACKAGE = "com.facebook.orca";
+    private static final String PLAY_STORE_LINK = "market://details?id=com.facebook.orca";
+    
+    public MessengerShare(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @Override
+    public void open(ReadableMap options) throws ActivityNotFoundException {
+        super.open(options);
+        //  extra params here
+        this.openIntentChooser();
+    }
+
+    @Override
+    protected String getPackage() {
+        return PACKAGE;
+    }
```

Comment:
It seems that the getPackage / getDefaultWebLink / getPlayStoreLink implementations here are all repeated and add no value.

Would it be simpler to add default implementations to SingleShareIntent, then in these objects simply set SingleShareIntent.appStoreURL and SingleShareIntent.playStoreURL correctly?